esp-hal-common/src/pcnt/mod.rs

```diff
@@ -1,4 +1,136 @@
 //! Pulse Counter peripheral driver
+//!
+//! ## Overview
+//! The `PCNT (Pulse Control)` driver for `ESP` chips is a software component
+//! that provides an interface for controlling and utilizing the `PCNT`
+//! peripheral. The `PCNT` peripheral is a hardware module available in `ESP`
+//! chips, which functions as a pulse counter and encoder. It is capable of
+//! counting pulses and monitoring changes in signal levels from external
+//! sources.
+//!
+//! The `PCNT` driver is designed to offer convenient and efficient access to
+//! the functionalities of the `PCNT` peripheral. It consists of two main
+//! modules:
+//!    * [channel]
+//!    * [unit]
+//!
+//! Each serving sprecific purposes.
```

Comment:
```suggestion
//! Each serving specific purposes.
```

This sentence looks weird in its own line to me.

Reply:
`PCNT` module documentation was edited as well, thank you!